cmd/scan/scan.go

```diff
@@ -0,0 +1,289 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"strings"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcutil"
+)
+
+type NewBTCAddress struct {
+	Addresses []string `json:"btc_addresses""`
+}
+
+type Address struct {
+	Addr         string `json:"address"`
+	MinScanBlock int64  `json:"min_scan_block"`
+	MidScanBlock int64  `json:"mid_scan_block"`
+	MaxScanBlock int64  `json:"max_scan_block"`
+	Txs          []Tx   `json:"txs"`
+}
+
+type Tx struct {
+	TxHash      string `json:"tx_hash"`
+	BlockHash   string `json:"block_hash"`
+	ParentHash  string `json:"parent_hash"`
+	BlockHeight int64  `json:"block_height"`
+}
+
+type Deposit struct {
+	Addr string
+	Tx   Tx
+}
+
+func ScanBlock(client *rpcclient.Client, blockID int64) ([]Deposit, error) {
+	blockHash, err := client.GetBlockHash(blockID)
+	if err != nil {
+		return nil, err
+	}
+	block, err := client.GetBlockVerboseTx(blockHash)
+	if err != nil {
+		return nil, err
+	}
+	parentHash := block.PreviousHash
+	var deposits []Deposit
+	var depTx Tx
+	depTx.BlockHeight = blockID
+	depTx.BlockHash = blockHash.String()
+	depTx.ParentHash = parentHash
+	for _, tx := range block.RawTx {
+		for i, addr := range tx.Vout {
+			depTx.TxHash = fmt.Sprintf("%s:%d", tx.Txid, i)
+
+			if len(addr.ScriptPubKey.Addresses) > 0 {
+				deposits = append(deposits, Deposit{
+					Addr: addr.ScriptPubKey.Addresses[0],
+					Tx:   depTx,
+				})
+			}
+		}
+
+	}
+	return deposits, nil
+}
+
+func CompareAddress(addr Address, deps []Deposit) Address {
```

Comment:
Rename this, "CompareAddress" does not sound like it modifies the address
Maybe change to
`func FindTxs(addr Address, deps []Deposit) []Tx`

Then later,
```
txs := FindTxs(addr, deps)
addrs[i].Txs = append(addr.Txs, txs...)
```

Reply:
good